### PR TITLE
Renaming of lists

### PR DESCRIPTION
**Pull request by AndersBillLinden**
This is a proposal of how to rename lists.

http://scratch.mit.edu/discuss/topic/271/
http://scratch.mit.edu/discuss/topic/301/
http://scratch.mit.edu/discuss/topic/42370/
http://scratch.mit.edu/discuss/topic/46622/
http://scratch.mit.edu/discuss/topic/52360/
http://scratch.mit.edu/discuss/topic/65530/
https://github.com/LLK/scratch-flash/issues/375

### CODE REVIEW

**Comment by AndersBillLinden** [comment]
This change will add the need of some translations.

**Comment by AndersBillLinden** [comment]
"You can only rename a list when pigs fly."

**Comment by AndersBillLinden** [comment]
Is the list rename functionality needed?

**Comment by sclements** [comment]
I think we should have it.

File: src/scratch/ScratchRuntime.as [hunk 1]
```diff
@@ -664,6 +664,39 @@ public class ScratchRuntime {
...
+		var owner:ScratchObj = app.viewedObj();
+		if (!owner.ownsList(oldName)) owner = app.stagePane;
+		if (owner.hasName(newName)) {
+			DialogBox.notify("Cannot Rename", "That name is already in use.");
+			return;
+		}
+
+		var v:ListWatcher = owner.lookupList(oldName);
+		if (v != null) {
+			v.listName = newName;
```
Comment (sclements) [inline src/scratch/ScratchRuntime.as]: Seems unnecessary since changeName() already updates the listName.

**Comment by sclements** [comment]
After making that minor change we'll test it and hopefully get it in soon. Thanks!

**Comment by AndersBillLinden** [comment]
I removed the extra line @sclements

**Comment by jwzimmer** [comment]
@AndersBillLinden, Thank you for working on this feature! However, we're not going to add this new feature to scratch-flash; as we focus on Scratch 3.0, we're focusing on developing new features there, and will consider adding this feature to our next major release. 
We appreciate that you have taken the time to contribute to Scratch! The scratch-blocks and scratch-vm repositories are really great places for open source contributions right now... Hope to see you there. : )
Thanks again!

**Pull request closed without merging**